src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,579 @@
+    """
+    The SuperPointInterestPointDecoder uses the output of the SuperPointEncoder to compute the keypoint with scores.
+    The scores are first computed by a convolutional layer, then a softmax is applied to get a probability distribution
+    over the 65 possible keypoint classes. The keypoints are then extracted from the scores by thresholding and
+    non-maximum suppression. Post-processing is then applied to remove keypoints too close to the image borders as well
+    as to keep only the k keypoints with highest score.
+    """
+
+    def __init__(self, config: SuperPointConfig) -> None:
+        super().__init__()
+        self.conv_layers_sizes = config.conv_layers_sizes
+        self.descriptor_dim = config.descriptor_dim
+        self.keypoint_threshold = config.keypoint_threshold
+        self.max_keypoints = config.max_keypoints
+        self.nms_radius = config.nms_radius
+        self.border_removal_distance = config.border_removal_distance
+
+        self.relu = nn.ReLU(inplace=True)
+        self.pool = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.convSa = nn.Conv2d(
+            self.conv_layers_sizes[3],
+            self.conv_layers_sizes[4],
+            kernel_size=3,
+            stride=1,
+            padding=1,
+        )
+        self.convSb = nn.Conv2d(self.conv_layers_sizes[4], 65, kernel_size=1, stride=1, padding=0)
+
+    def forward(self, encoded: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        # Compute the dense keypoint scores
+        scores = self.get_scores(encoded)
+        # Extract keypoints
+        keypoints, scores = self.extract_keypoints(scores)
+
+        return keypoints, scores
+
+    def get_scores(self, encoded: torch.Tensor) -> torch.Tensor:
+        """Compute the dense keypoint scores"""
+        scores = self.relu(self.convSa(encoded))
+        scores = self.convSb(scores)
+        scores = torch.nn.functional.softmax(scores, 1)[:, :-1]
+        batch_size, _, height, width = scores.shape
+        scores = scores.permute(0, 2, 3, 1).reshape(batch_size, height, width, 8, 8)
+        scores = scores.permute(0, 1, 3, 2, 4).reshape(batch_size, height * 8, width * 8)
+        scores = self.simple_nms(scores, self.nms_radius)
+        return scores
+
+    def extract_keypoints(self, scores: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        _, height, width = scores.shape
+
+        # Threshold keypoints by score value
+        keypoints = torch.nonzero(scores[0] > self.keypoint_threshold)
+        scores = scores[0][tuple(keypoints.t())]
+
+        # Discard keypoints near the image borders
+        keypoints, scores = self.remove_borders(keypoints, scores, self.border_removal_distance, height * 8, width * 8)
+
+        # Keep the k keypoints with highest score
+        if self.max_keypoints >= 0:
+            keypoints, scores = self.top_k_keypoints(keypoints, scores, self.max_keypoints)
+
+        # Convert (h, w) to (x, y)
```

Comment:
Here `height` and `width` are the the height and width of the input image, since scores is a tensor providing a score for each pixel in the given image (see comments on line 185)

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/148d1ae81874d779803a1784bb276e61cbef4a4c)